snapcraft/internal/meta/_snap_packaging.py

```diff
@@ -133,6 +133,97 @@ def _update_yaml_with_extracted_metadata(
         raise meta_errors.MissingSnapcraftYamlKeysError(keys=missing_keys)
 
 
+def _adopt_info(
+        config_data: Dict[str, Any],
+        extracted_metadata: _metadata.ExtractedMetadata):
+    metadata_dict = extracted_metadata.to_dict()
+    for key, value in filter(
+            lambda item: item != 'desktop',
```

Comment:
Personally, I'd rather just see an `if` statement in the loop instead of this fancy thing :stuck_out_tongue: . It causes a double-take for something that is otherwise a simple one-liner.

Reply:
I sort of like functional steps so I don't mind either way :-)

If we didn't have the 80 char limit I would be rooting for this :100: though :wink:. Alas I will let @kyrofa has his way :smile:

Reply:
I might have gone a little too far with the itertools ^_^